File: problems/1. Two Sum/Solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        table = {}
        for i in range(len(nums)):
            complement = target - nums[i]
            if complement in table.keys() and table[complement] != i:
                return [i, table[complement]]
            table[nums[i]] = i
        return []

File: problems/1. Two Sum/Solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::map<int, int> map;
        for (int i = 0; i < nums.size(); i++) {
            int complement = target - nums[i];
            if (map.count(complement) > 0 && map[complement] != i) {
                return {map[complement], i};
            }
            map[nums[i]] = i;
        }
        return {0, 0};
    }
};